app/models/manageiq/providers/openshift/container_manager/refresh_parser.rb

```diff
@@ -172,29 +172,33 @@ def parse_env_variables(env_variables)
       end
 
       def parse_openshift_image(openshift_image)
-        id = openshift_image[:dockerImageReference]
-        ref = ContainerImage::DOCKER_PULLABLE_PREFIX + id
+        id = openshift_image[:dockerImageReference] || openshift_image[:metadata][:name]
+        ref = ContainerImage::DOCKER_PULLABLE_PREFIX + id.to_s
         new_result = parse_container_image(id, ref)
 
-        json = JSON.parse(openshift_image[:dockerImageManifest])
-        new_result[:tag] ||= json["tag"] if json.keys.include?("tag")
+        if openshift_image[:dockerImageManifest].present?
+          json = JSON.parse(openshift_image[:dockerImageManifest])
```

Comment:
@zeari @Fryguy I agree that the JSON payload can hardly be malformed but this part is so critical (not crashing the inventory refresh) that it may deserve some extra care (so that we won't have to deal with this in the future).
In case we can't parse the JSON string we could log an error and continue.